lib/ex_azure_key_vault.ex

```diff
@@ -55,6 +56,108 @@ defmodule ExAzureKeyVault.Client do
     }
   end
 
+  @doc """
+  Connects to Azure Key Vault (Using MSI - Managed Service Identity).
+
+  ## Examples
+
+  When defining environment variables and/or adding to configuration.
+
+      $ export AZURE_VAULT_NAME="my-vault"
+
+      # Config.exs
+      config :ex_azure_key_vault,
+        azure_vault_name: {:system, "AZURE_VAULT_NAME"}
+
+      iex(1)> ExAzureKeyVault.Client.msi_connect()
+      %ExAzureKeyVault.Client{
+        api_version: "2016-10-01",
+        bearer_token: "Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9...",
+        vault_name: "my-vault"
+      }
+
+  Passing custom parameters.
+
+      iex(1)> ExAzureKeyVault.Client.msi_connect("custom-vault")
+      %ExAzureKeyVault.Client{
+        api_version: "2016-10-01",
+        bearer_token: "Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9...",
+        vault_name: "custom-vault"
+      }
+
+  """
+  @spec smart_connect() :: Client.t() | {:error, any}
```

Comment:
This looks out of scope for this feature. Could you move this piece to a separate PR for further discussion?